### ISSUE TITLE
PyO3 has a build dependency on Python 2

### ISSUE THREAD

**Issue by vadixidav** [PyO3/pyo3#584]
### 🌍 Environment

```Dockerfile
FROM ubuntu:18.04 as build

# Install dependencies.
RUN set -eux; \
    apt-get update \
    && apt-get install -y --no-install-recommends \
    pkg-config \
    libssl-dev \
    curl \
    ca-certificates \
    gcc \
    libc-dev \
    python3-dev \
    && rm -rf /var/lib/apt/lists/*
ENV LANG en_US.utf8

# Rust
RUN curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly-2019-08-28-x86_64-unknown-linux-gnu -y
ENV PATH "/root/.cargo/bin:$PATH"
```

### 💥 Reproducing

Add this dependency:

`pyo3 = "0.7.0"`

Now build your code. You will see the following:

```
error: failed to run custom build command for `pyo3 v0.7.0`

Caused by:
  process didn't exit successfully: `/web/target/release/build/pyo3-23c79148396499f4/build-script-build` (exit code: 1)
--- stderr
Error: "Could not find any interpreter at python, are you sure you have python installed on your PATH?"
```

### Workaround

I was able to fix this temporarily by adding the following to my Dockerfile:

```Dockerfile
# Force python to be python3.
RUN rm -f /usr/bin/python && ln -s /usr/bin/python3 /usr/bin/python
```

I notice that on the main `README.md` it is stated to run the following on Ubuntu:

```bash
sudo apt install python3-dev python-dev
```

However, I would prefer to only install `python3-dev` to minimize downloads, build time, etc for Docker. I feel like if just linking python to python3 works, it should probably also support discovering python3 and running python3 on the system.

Basically, another way to word this ticket is "Remove build dependency on Python 2".

**Comment by kngwyu** [comment]
Closed via #596
